data-pipeline/src/trace_exporter.rs

```diff
@@ -214,32 +253,173 @@ impl TraceExporter {
 
     /// Safely shutdown the TraceExporter and all related tasks
     pub fn shutdown(self, timeout: Option<Duration>) -> Result<(), String> {
-        match self.stats {
-            StatsComputationStatus::StatsEnabled {
-                stats_concentrator: _,
-                cancellation_token: cancelation_token,
-                exporter_handle,
-            } => {
-                if let Some(timeout) = timeout {
-                    match self.runtime.block_on(async {
-                        tokio::time::timeout(timeout, async {
-                            cancelation_token.cancel();
-                            let _ = exporter_handle.await;
-                        })
-                        .await
-                    }) {
-                        Ok(_) => Ok(()),
-                        Err(_) => Err("Shutdown timed out".to_string()),
-                    }
-                } else {
-                    self.runtime.block_on(async {
-                        cancelation_token.cancel();
+        if let Some(timeout) = timeout {
+            match self.runtime.block_on(async {
+                tokio::time::timeout(timeout, async {
+                    let stats_status: Option<StatsComputationStatus> =
+                        std::sync::Arc::<StatsComputationStatus>::into_inner(
+                            self.client_side_stats.into_inner(),
+                        );
+                    if let Some(StatsComputationStatus::Enabled {
+                        stats_concentrator: _,
+                        cancellation_token,
+                        exporter_handle,
+                    }) = stats_status
+                    {
+                        cancellation_token.cancel();
                         let _ = exporter_handle.await;
-                    });
-                    Ok(())
+                    }
+                })
+                .await
+            }) {
+                Ok(()) => Ok(()),
+                Err(_) => Err("Shutdown timed out".to_string()),
+            }
+        } else {
+            self.runtime.block_on(async {
+                let stats_status: Option<StatsComputationStatus> =
+                    std::sync::Arc::<StatsComputationStatus>::into_inner(
+                        self.client_side_stats.into_inner(),
+                    );
+                if let Some(StatsComputationStatus::Enabled {
+                    stats_concentrator: _,
+                    cancellation_token,
+                    exporter_handle,
+                }) = stats_status
+                {
+                    cancellation_token.cancel();
+                    let _ = exporter_handle.await;
+                }
+            });
+            Ok(())
+        }
+    }
+
+    /// Start the stats exporter and enable stats computation
+    ///
+    /// Should only be used if the agent enabled stats computation
```

Comment:
Should we add a note here that this won't work if stats computation is disabled at the client level?

Reply:
I think it's fine if the the client didn't enable stats computation there's no point for the exporter to start it and since it is private I don't think there's a risk of misusing it